From 410f8296638dec9dccdb65927a77baa40fbd5cc4 Mon Sep 17 00:00:00 2001
From: Tim Gover <tim.gover@raspberrypi.org>
Date: Fri, 22 Jun 2018 14:11:27 +0100
Subject: [PATCH 189/220] ASoC: hifiberry_api: Move snd_soc_dai_set_bclk_ratio
 to .init

snd_soc_dai_set_bclk_ratio is called with a fixed value of 64.
This can be called from .init instead allowing the more verbose
hw_params callback method to be removed.

Signed-off-by: Tim Gover <tim.gover@raspberrypi.org>
---
 sound/soc/bcm/hifiberry_amp.c | 17 +----------------
 1 file changed, 1 insertion(+), 16 deletions(-)

diff --git a/sound/soc/bcm/hifiberry_amp.c b/sound/soc/bcm/hifiberry_amp.c
index 7d9d17c..3b0f00e 100644
--- a/sound/soc/bcm/hifiberry_amp.c
+++ b/sound/soc/bcm/hifiberry_amp.c
@@ -25,23 +25,9 @@
 
 static int snd_rpi_hifiberry_amp_init(struct snd_soc_pcm_runtime *rtd)
 {
-	// ToDo: init of the dsp-registers.
-	return 0;
+	return snd_soc_dai_set_bclk_ratio(rtd->cpu_dai, 64);
 }
 
-static int snd_rpi_hifiberry_amp_hw_params( struct snd_pcm_substream *substream,
-				       struct snd_pcm_hw_params *params )
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_soc_dai *cpu_dai = rtd->cpu_dai;
-
-	return snd_soc_dai_set_bclk_ratio(cpu_dai, 64);
-}
-
-static struct snd_soc_ops snd_rpi_hifiberry_amp_ops = {
-	.hw_params = snd_rpi_hifiberry_amp_hw_params,
-};
-
 static struct snd_soc_dai_link snd_rpi_hifiberry_amp_dai[] = {
     {
 		.name			= "HifiBerry AMP",
@@ -53,7 +39,6 @@ static int snd_rpi_hifiberry_amp_hw_params( struct snd_pcm_substream *substream,
 		.dai_fmt		= SND_SOC_DAIFMT_I2S |
 						  SND_SOC_DAIFMT_NB_NF |
 						  SND_SOC_DAIFMT_CBS_CFS,
-		.ops			= &snd_rpi_hifiberry_amp_ops,
 		.init			= snd_rpi_hifiberry_amp_init,
 	},
 };
-- 
1.9.1

